File: quest/arrays/03_max_consecutive.py
class Solution(object):
    def raw_findMaxConsecutiveOnes(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """

        ones = 0
        tmp = 0

        for x in nums:
            if x == 1:
                tmp += 1
                if tmp > ones:
                    ones = tmp
            else:
                tmp = 0

        return ones

    def findMaxConsecutiveOnes(self, nums):
        ones = 0
        tmp = 0

        for x in nums:
            if x == 1:
                tmp += 1
                ones = max(ones, tmp)
            else:
                tmp = 0

        return ones

File: quest/arrays/03_max_consecutive.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findMaxConsecutiveOnes = function(nums) {
    var ones = 0;
    var tmp = 0;

    for (i = 0; i < nums.length; i++) {
        if (nums[i] == 1) {
            tmp += 1;

            if (tmp > ones) {
                ones = tmp;
            }
        } else {
            tmp = 0;
        }
    }

    return ones;
};
